libs/core/langchain_core/tools/structured.py

```diff
@@ -41,6 +42,9 @@ class StructuredTool(BaseTool):
     """The function to run when the tool is called."""
     coroutine: Optional[Callable[..., Awaitable[Any]]] = None
     """The asynchronous version of the function."""
+    outer_instance: Optional[Any] = None
```

Comment:
Let's try to avoid storing this if possible. Could we instead set `.func` to a callable that handles the plumbing for the user?

Reply:
(in other words, let's not modify `StructuredTool`)

Reply:
Yes